repository/src/main/java/org/apache/atlas/tasks/TaskQueueWatcher.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class TaskQueueWatcher implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(TaskQueueWatcher.class);
+    private static final TaskExecutor.TaskLogger TASK_LOG         = TaskExecutor.TaskLogger.getLogger();
+
+    private TaskRegistry registry;
+    private final ExecutorService executorService;
+    private final Map<String, TaskFactory> taskTypeFactoryMap;
+    private final TaskManagement.Statistics statistics;
+
+    private static long pollInterval = AtlasConfiguration.TASKS_REQUEUE_POLL_INTERVAL.getLong();
+
+    private CountDownLatch latch = null;
+
+    private final AtomicBoolean shouldRun = new AtomicBoolean(false);
+
+    public TaskQueueWatcher(ExecutorService executorService, TaskRegistry registry,
+                            Map<String, TaskFactory> taskTypeFactoryMap, TaskManagement.Statistics statistics,
+                            CountDownLatch latch) {
+
+        this.registry = registry;
+        this.executorService = executorService;
+        this.taskTypeFactoryMap = taskTypeFactoryMap;
+        this.statistics = statistics;
+        this.latch = latch;
+    }
+
+    public void shutdown() {
+        shouldRun.set(false);
+        LOG.info("TaskQueueWatcher: Shutdown");
+    }
+
+    @Override
+    public void run() {
+        shouldRun.set(true);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("TaskQueueWatcher: running {}:{}", Thread.currentThread().getName(), Thread.currentThread().getId());
+        }
+
+        while (shouldRun.get()) {
+            try {
+                if (latch != null && latch.getCount() != 0) {
+                    LOG.info("TaskQueueWatcher: Waiting on Latch, current count: {}", latch.getCount());
+                    latch.await();
+                }
+
+                if (latch != null) {
+                    LOG.info("TaskQueueWatcher: Latch wait complete!!");
+                }
+
+                TasksFetcher fetcher = new TasksFetcher(registry);
+
+                Thread tasksFetcherThread = new Thread(fetcher);
+                tasksFetcherThread.start();
+                tasksFetcherThread.join();
+
+                List<AtlasTask> tasks = fetcher.getTasks();
+                if (CollectionUtils.isNotEmpty(tasks)) {
+                    addAll(tasks);
+                } else {
+                    LOG.info("No tasks to queue, sleeping for {} ms", pollInterval);
+                }
+
+                Thread.sleep(pollInterval);
+
+            } catch (InterruptedException interruptedException) {
+                LOG.error("TaskQueueWatcher: Interrupted: thread is terminated, new tasks will not be loaded into the queue until next restart");
+                break;
+            } catch (Exception e){
+                LOG.error("TaskQueueWatcher: Exception occurred");
```

Comment:
pass exception object for better tracing of exception. 
`LOG.error("TaskQueueWatcher: Exception occurred", e);`
can removed e.printStackTrace